src/main/java/org/dita/dost/writer/NormalizeTableFilter.java

```diff
@@ -142,48 +141,56 @@ private void generateColSpec() throws SAXException {
+                    }
                 }
+            } else {
+                prev = new Span(1, 1);
             }
-        } else {
-            prev = new Span(1, 1);
-        }
-        grow(tableState.currentRow, tableState.currentColumn + colspan);
-        final Span span = new Span(colspan, rowspan);
+            grow(tableState.currentRow, tableState.currentColumn + colspan);
+            final Span span = new Span(colspan, rowspan);
 
-        tableState.currentRow.set(tableState.currentColumn, span);
+            tableState.currentRow.set(tableState.currentColumn, span);
 
-        XMLUtils.addOrSetAttribute(res, ATTRIBUTE_NAME_COLNAME, COLUMN_NAME_COL + (tableState.currentColumn + 1));
-        if (res.getValue(ATTRIBUTE_NAME_NAMEST) != null) {
-            XMLUtils.addOrSetAttribute(res, ATTRIBUTE_NAME_NAMEST, COLUMN_NAME_COL + tableState.columnNumber);
-        }
-        if (res.getValue(ATTRIBUTE_NAME_NAMEEND) != null) {
-            XMLUtils.addOrSetAttribute(res, ATTRIBUTE_NAME_NAMEEND, COLUMN_NAME_COL + getEndNumber(res, tableState.columnNumber));
-            XMLUtils.addOrSetAttribute(res, DITA_OT_NS, ATTR_MORECOLS, DITA_OT_NS_PREFIX + ":" + ATTR_MORECOLS, "CDATA", Integer.toString(span.x - 1));
-        }
-        // Add extensions
-        XMLUtils.addOrSetAttribute(res, DITA_OT_NS, ATTR_X, DITA_OT_NS_PREFIX + ":" + ATTR_X, "CDATA", Integer.toString(tableState.currentColumn + 1));
-        XMLUtils.addOrSetAttribute(res, DITA_OT_NS, ATTR_Y, DITA_OT_NS_PREFIX + ":" + ATTR_Y, "CDATA", Integer.toString(tableState.rowNumber));
+            XMLUtils.addOrSetAttribute(res, ATTRIBUTE_NAME_COLNAME, COLUMN_NAME_COL + (tableState.currentColumn + 1));
+            if (res.getValue(ATTRIBUTE_NAME_NAMEST) != null) {
+                XMLUtils.addOrSetAttribute(res, ATTRIBUTE_NAME_NAMEST, COLUMN_NAME_COL + tableState.columnNumber);
+            }
+            if (res.getValue(ATTRIBUTE_NAME_NAMEEND) != null) {
+                XMLUtils.addOrSetAttribute(res, ATTRIBUTE_NAME_NAMEEND, COLUMN_NAME_COL + getEndNumber(res, tableState.columnNumber));
+                XMLUtils.addOrSetAttribute(res, DITA_OT_NS, ATTR_MORECOLS, DITA_OT_NS_PREFIX + ":" + ATTR_MORECOLS, "CDATA", Integer.toString(span.x - 1));
+            }
+            // Add extensions
+            XMLUtils.addOrSetAttribute(res, DITA_OT_NS, ATTR_X, DITA_OT_NS_PREFIX + ":" + ATTR_X, "CDATA", Integer.toString(tableState.currentColumn + 1));
+            XMLUtils.addOrSetAttribute(res, DITA_OT_NS, ATTR_Y, DITA_OT_NS_PREFIX + ":" + ATTR_Y, "CDATA", Integer.toString(tableState.rowNumber));
 
-        tableState.currentColumn = tableState.currentColumn + colspan;
-        tableState.columnNumberEnd = getEndNumber(res, tableState.columnNumber);
+            tableState.currentColumn = tableState.currentColumn + colspan;
+            tableState.columnNumberEnd = getEndNumber(res, tableState.columnNumber);
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
This is kinda shotgun solution to broken tables that may cause exceptions in the normalization code. But it's better than failing or not logging anything.

Reply:
Why not add an extra stop condition in the for loop where the problem originates instead of catching a runtime exception?